cmd/controller-manager/app/controller_manager.go

```diff
@@ -111,21 +112,30 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 		rest.AddUserAgent(k8sKubeconfig, controllerManagerAgentName),
 	)
 	if err != nil {
-		glog.Fatalf("Invalid k8s API configuration: %v", err)
+		return fmt.Errorf("invalid Kubernetes API configuration: %v", err)
```

Comment:
@simonleung8 Thinking on this more, `glog.Fatal` seems like it is more equivalent to `panic` than exiting, since it spews the stack traces.  So returning the error definitely seems like the correct approach here.